exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -98,99 +96,74 @@
-        frameworkInjector);
-  }
-
-  private static TestKit newInstanceWithInjector(
-      List<Class<? extends ServiceModule>> serviceModules, EmulatedNodeType nodeType,
-      short validatorCount, @Nullable TimeProvider timeProvider, Injector frameworkInjector) {
-    Map<Short, UserServiceAdapter> serviceAdapters = toUserServiceAdapters(
-        serviceModules, frameworkInjector);
+                                     @Nullable TimeServiceSpec timeServiceSpec) {
     boolean isAuditorNode = nodeType == EmulatedNodeType.AUDITOR;
-    UserServiceAdapter[] userServiceAdapters = serviceAdapters.values()
-        .toArray(new UserServiceAdapter[0]);
-    TimeProviderAdapter timeProviderAdapter =  timeProvider == null
-        ? null
-        : new TimeProviderAdapter(timeProvider);
-    long nativeHandle = nativeCreateTestKit(userServiceAdapters, isAuditorNode, validatorCount,
-        timeProviderAdapter);
-    return new TestKit(nativeHandle, serviceAdapters);
+    long nativeHandle = nativeCreateTestKit(
+        serviceInstances.toArray(new TestKitServiceInstances[0]), isAuditorNode, validatorCount,
+        timeServiceSpec);
+    return new TestKit(nativeHandle, timeServiceSpec);
   }
 
   /**
-   * Returns a list of user service adapters created from given service modules.
+   * Deploys and creates a single service with a single validator node in this TestKit network.
    */
-  private static Map<Short, UserServiceAdapter> toUserServiceAdapters(
-      List<Class<? extends ServiceModule>> serviceModules, Injector frameworkInjector) {
-    List<UserServiceAdapter> services = serviceModules.stream()
-        .map(s -> createUserServiceAdapter(s, frameworkInjector))
-        .collect(toList());
-    // UserServiceAdapters are removed, so this usage is a false signal for keeping getId()
-    return Maps.uniqueIndex(services, UserServiceAdapter::getId);
-  }
-
-  /**
-   * Instantiates a service given its module and wraps in a UserServiceAdapter for the native code.
-   */
-  private static UserServiceAdapter createUserServiceAdapter(
-      Class<? extends ServiceModule> moduleClass, Injector frameworkInjector) {
-    try {
-      Supplier<ServiceModule> moduleSupplier = new ReflectiveModuleSupplier(moduleClass);
-      ServiceModule serviceModule = moduleSupplier.get();
-      Injector serviceInjector = frameworkInjector.createChildInjector(serviceModule);
-      return serviceInjector.getInstance(UserServiceAdapter.class);
-    } catch (IllegalAccessException e) {
-      throw new IllegalArgumentException("Cannot access the no-arg module constructor", e);
-    } catch (NoSuchMethodException e) {
-      throw new IllegalArgumentException("No no-arg constructor", e);
-    }
+  public static TestKit forService(ServiceArtifactId artifactId, String artifactFilename,
+                                   String serviceName, int serviceId, Any configuration) {
```

Comment:
Any is not used for configuration anymore, but any protobuf message. This method may accept it as MessageLite.